newswires/client/src/App.tsx

```diff
@@ -1,9 +1,229 @@
+import { css } from '@emotion/react';
+import { useCallback, useEffect, useMemo, useState } from 'react';
+import sanitizeHtml from 'sanitize-html';
 import './App.css';
 
+type WireData = {
+	id: number;
+	externalId: string;
+	ingestedAt: string;
+	content: Partial<{
+		uri: string;
+		usn: string;
+		version: string;
+		firstVersion: string; // date
+		versionCreated: string; // date
+		dateTimeSent: string; //date
+		headline: string;
+		subhead: string;
+		byline: string;
+		keywords: string;
+		usage: string;
+		location: string;
+		body_text: string;
+	}>;
+};
+
+type PageStage = { loading: true } | { error: string } | WireData[];
+
+const querify = (query: string): string => {
+	if (query.trim().length <= 0) return '';
+	const params = new URLSearchParams();
+	params.set('q', query.trim());
+	return '?' + params.toString();
+};
+
+// eslint-disable-next-line @typescript-eslint/no-explicit-any -- suitably generic function
+const debounce = <F extends (...args: any[]) => void>(
+	f: F,
+	delay: number,
+): ((...args: Parameters<F>) => void) => {
+	let waiting: ReturnType<typeof setTimeout> | undefined;
+
+	return (...args: Parameters<F>) => {
+		if (waiting !== undefined) {
+			clearTimeout(waiting);
+		}
+		waiting = setTimeout(() => f(...args), delay);
+	};
+};
+
 export function App() {
+	const [pageState, setPageState] = useState<PageStage>({ loading: true });
+
+	const [query, setQuery] = useState<string>('');
+
+	const [selected, setSelected] = useState<WireData | undefined>(undefined);
+
+	const safeBodyText = useMemo(() => {
+		return selected?.content.body_text
+			? sanitizeHtml(selected.content.body_text)
+			: undefined;
+	}, [selected]);
```

Comment:
we probably ought to add quite a strict Content Security Policy (like we have for https://github.com/guardian/editorial-wires) before this hits production, to guard against malicious content (e.g. XSS) that might find its way into wire content (as it's presented on our domain). we should consider baking in the hash of our own JS bundle such that it's the only JS allowed to run on the page.

Reply:
sounds good to me! That is exactly why I pass the content through `sanitize-html` first, but no reason not to do both - let's do this in a followup PR though